### PR TITLE
Interview scheduler

### PR DESCRIPTION
**Pull request by neverwannafly**
# Features Introduced
### Upload Resume from Profile
Screenshot 2020-05-18 at 08 00 42

### User Interviews
Screenshot 2020-05-18 at 08 00 56

### Single Interview Details
Screenshot 2020-05-18 at 08 01 11

### Edit Interview
Screenshot 2020-05-18 at 09 16 42

### Automated Update and Reminder Mails
Screenshot 2020-05-18 at 09 53 26
Screenshot 2020-05-18 at 09 55 29
Screenshot 2020-05-18 at 09 58 13

### CODE REVIEW

File: interview-scheduler/app/controllers/interviews_controller.rb [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+      }
     else
       render json: {}
     end
   end
 
+  # get interviews of a particular user
+  def user_interviews
+    if can_view
+      @interviews = Interview.where(:id => UserInterview.where(:user_id => current_user).pluck(:interview_id))
```
Comment (abhishek-8) [inline interview-scheduler/app/controllers/interviews_controller.rb:44]: We can join user_interviews and interviews here and get interviews in single query

File: interview-scheduler/app/controllers/interviews_controller.rb [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+    else
+      render json: {}
+    end
+  end
+
+  # Get a single interview
+  def get
+    if can_view
+      interview = Interview.find(params[:id])
+      members = User.where(:id => UserInterview.where(:interview_id => interview.id).pluck(:user_id)).pluck(:username ).join ","
```
Comment (abhishek-8) [inline interview-scheduler/app/controllers/interviews_controller.rb:26]: Join users and users_interviews. Avoid multiple db queries.

File: interview-scheduler/app/controllers/interviews_controller.rb [hunk 3]
```diff
@@ -48,7 +48,12 @@ def user_interviews
   end
 
   def update
+    
+  end
+
+  def edit
     @interview = Interview.find(params[:id])
+    @members = User.where(:id => UserInterview.where(:interview_id => @interview.id).pluck(:user_id)).pluck(:username ).join ","
```
Comment (abhishek-8) [inline interview-scheduler/app/controllers/interviews_controller.rb]: Joins can be used here also

File: interview-scheduler/app/controllers/interviews_controller.rb [hunk 4]
```diff
@@ -53,8 +127,10 @@ def check_conflicts(members)
...
     interviews = UserInterview.where(user_id: members)
     interviews.each do |interview|
-      query = Interview.where(:id => interview.interview_id, :start => start_time..end_time).or(Interview.where(:id => interview.interview_id, :end => start_time..end_time-1.minutes))
-      if query.count != 0
+      upper_threshold = Time.now + 1000.years
+      lower_threshold = Time.now - 1000.years
+      query = Interview.where(:id => interview.interview_id, :start => end_time..upper_threshold).or(Interview.where(:id => interview.interview_id, :end => lower_threshold..start_time))
+      if query.count != Interview.where(:id=>interview.interview_id).count
         conflicts = true
         break
```
Comment (ayu-111) [inline interview-scheduler/app/controllers/interviews_controller.rb:135]: @neverwannafly Is this the best way to check for conflicts?

**Pull request merged**